modin/backends/base/query_compiler.py

```diff
@@ -1182,3 +1203,7 @@ def delitem(self, key):
         return self.drop(columns=[key])
 
     # END __delitem__
+
+    @abc.abstractmethod
+    def has_multiindex(self):
```

Comment:
This will be good to put in a stand-alone PR. As a part of this change, we can change everything in the API layer where we are checking `isinstance(self.index, pandas.MultiIndex)`.